### ISSUE TITLE
UX Improvements list

### ISSUE THREAD

**Issue by barryvdh** [barryvdh/laravel-debugbar#1654]
- Hide inactive tabs until used; https://github.com/maximebf/php-debugbar/pull/672
- Combine Route / Request tabs and make it easier to read? Also remove the tab or indicator, as it's double now, so instead move the tab to the right? 
- Maybe hide RAM en PHP version by default, by creating an info/performance tab?  Ram is a bit tricky, I never really pay much attention to it but can see it being useful.
- Improve Gate to show Policy or target; https://github.com/barryvdh/laravel-debugbar/pull/1653
- Improve View and Livewire tab for Liveware components
- Prettify querys by formatting whitespace on open?
- Check consistency in filters/links to IDE
- ...?

**Comment by barryvdh** [comment]
I like how Symfony shows the details on hover:
![image](https://github.com/user-attachments/assets/ed57adc4-3bda-47e9-82a9-0407f436298c)

**Comment by systemsolutionweb** [comment]
What about CacheCollector :

- #1576
- #1563

**Comment by barryvdh** [comment]
CacheCollector I'll look into,  but want to minimize performance impact.

Additionally, that popup from previous reply could also be used to toggle options.

**Comment by tpetry** [comment]
> Prettify querys by formatting whitespace on open?
I was thinking about that too. But common eloquent queries with multiple `has` or `whereHas` conditions get really long. And when prettifying those subqueries they can easily span 50+ rows.

**Comment by barryvdh** [comment]
> > Prettify querys by formatting whitespace on open?
> 
> I was thinking about that too. But common eloquent queries with multiple `has` or `whereHas` conditions get really long. And when prettifying those subqueries they can easily span 50+ rows.

Yes I wanted to do it more like the 'Explain' button. So either Open and prettify when not collapsed, or open + button to prettify.

**Comment by parallels999** [comment]
>open + button to prettify

maybe a popup window

**Comment by azgooon** [comment]
In the "Route" tab, when a page containing further Ajax calls is loaded, the debug bar detects these requests and displays the latest one which I find often not the one I am working on. Having the ability to disable auto-preselecting new requests would be handy.  

![image](https://github.com/user-attachments/assets/7a2fafbc-b188-4791-9a48-e073582aaa3f)

What I mean is that every time the page loads I need to go to the list of request and select the first one (as I am working on the main document)

**Comment by parallels999** [comment]
@azgooon hi, that is not the point of this thread,
but I don't know if the documentation for that is wrong, they ask the same thing every time
https://github.com/barryvdh/laravel-debugbar/blob/16a13cc5221aee90ae20aa59083ced2211e714eb/config/debugbar.php#L128

**Comment by barryvdh** [comment]
And if you click the button, you can disable it from the UI (autoshow)
![image](https://github.com/user-attachments/assets/5ca63116-c1c9-4dbb-a2dc-49efe5036347)

**Comment by angeljqv** [comment]
> > > Prettify querys by formatting whitespace on open?
> > 
> > 
> > I was thinking about that too. But common eloquent queries with multiple `has` or `whereHas` conditions get really long. And when prettifying those subqueries they can easily span 50+ rows.
> 
> Yes I wanted to do it more like the 'Explain' button. So either Open and prettify when not collapsed, or open + button to prettify.

+1 for the pop up, 
maybe repo: [sql-formatter-org/sql-formatter](https://github.com/sql-formatter-org/sql-formatter), js: [@15.4.5/dist/sql-formatter.min.js](https://cdn.jsdelivr.net/npm/sql-formatter@15.4.5/dist/sql-formatter.min.js), demo: [/sql-formatter](https://sql-formatter-org.github.io/sql-formatter/)

**Comment by erikn69** [comment]
>repo: [sql-formatter-org/sql-formatter](https://github.com/sql-formatter-org/sql-formatter), js: [@15.4.5/dist/sql-formatter.min.js](https://cdn.jsdelivr.net/npm/sql-formatter@15.4.5/dist/sql-formatter.min.js), demo: [/sql-formatter](https://sql-formatter-org.github.io/sql-formatter/)

I did made a popup PoC on https://github.com/maximebf/php-debugbar/pull/690, this would be the result:
![image](https://github.com/user-attachments/assets/0c7a2e70-3616-4a17-a22a-03d2f66d180d)

![image](https://github.com/user-attachments/assets/c489a9ca-a646-4dc8-87ed-0d0fdc99bdf5)

**Comment by bilogic** [comment]
Possible to implement `addIndicator` for "Custom tabs and indicators" described here http://phpdebugbar.com/docs/javascript-bar.html#tabs-and-indicators

Would be fantastic to add my own custom links to
1. https://github.com/opcodesio/log-viewer
2. https://github.com/apih/laravel-route-list-web
3. phpinfo()

**Comment by josephbradfield** [comment]
I know there is a select for determining the position, as far as i can tell we cannot set that in the config file `DEBUGBAR_POSITION=bottomRight` would be nice. I understand that it writes to localStorage but in my app it was overlaying on my user login button or hiding other important pieces if the user hasn't set and stored that yet so I used this piece to css override it and force to right even if left is selected. 
```css
.phpdebugbar[data-openbtnposition="bottomLeft"].phpdebugbar-closed {
    /*override bottomLeft to force it to right side*/
    left: auto;
    right: 0;
}
```
